libs/server/InputHeader.cs

```diff
@@ -358,22 +333,20 @@ public RawStringInput(ushort cmd, byte flags = 0, long arg1 = 0) :
         /// </summary>
         /// <param name="cmd">Command</param>
         /// <param name="parseState">Parse state</param>
-        /// <param name="parseStateFirstArgIdx">Index at which to start reading parse state parameters array</param>
-        /// <param name="parseStateLastArgIdx">Last index to read in the parse state parameters array for command execution</param>
+        /// <param name="startIdx">First command argument index in parse state</param>
+        /// <param name="argCount">Command argument count in parse state (default: -1 for all arguments available starting at startIdx)</param>
         /// <param name="arg1">General-purpose argument</param>
         /// <param name="flags">Flags</param>
-        public RawStringInput(RespCommand cmd, ref SessionParseState parseState, int parseStateFirstArgIdx = 0,
-            int parseStateLastArgIdx = -1, long arg1 = 0, RespInputFlags flags = 0) : this(cmd, flags, arg1)
+        public RawStringInput(RespCommand cmd, ref SessionParseState parseState, int startIdx = 0, int argCount = -1, long arg1 = 0, RespInputFlags flags = 0) : this(cmd, flags, arg1)
         {
             this.parseState = parseState;
-            this.parseStateFirstArgIdx = parseStateFirstArgIdx;
-            this.parseStateLastArgIdx = parseStateLastArgIdx;
+            this.parseState.Slice(startIdx, argCount);
```

Comment:
this.parseState = parseState.Slice(...) - i dont think we should be modifying fields of already created structs. These fields are to be treated as effectively read-only.